app/(app)/(purchase-and-payment)/ticket-purchase.tsx

```diff
@@ -19,27 +19,25 @@ export type TicketPurchaseSearchParams = {
 
 /**
  * This page takes care of redirect from Payment gate. Do not change its path, unless it's changes on BE too.
- *
- * @constructor
  */
-
-// TODO repeat request to BE until the state of ticket is SUCCESS or FAIL
 const TicketPurchasePage = () => {
   const t = useTranslation('PurchaseScreen')
   const { ticketId } = useLocalSearchParams<TicketPurchaseSearchParams>()
   const ticketIdParsed = ticketId ? parseInt(ticketId, 10) : undefined
   const onPurchaseStoreUpdate = usePurchaseStoreUpdateContext()
 
-  const { data, isPending, isError, error } = useQueryWithFocusRefetch(
+  const { data, isPending, isError, error, refetch } = useQueryWithFocusRefetch(
     getTicketOptions(ticketIdParsed!),
   )
 
   useEffect(() => {
-    // TODO check if ticket is paid and reset store
-    if (data) {
+    if (data?.paymentStatus === 'SUCCESS') {
       onPurchaseStoreUpdate(defaultInitialPurchaseStoreValues)
+    } else if (data?.paymentStatus === 'PENDING') {
+      refetch()
     }
-  }, [data, onPurchaseStoreUpdate])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data])
```

Comment:
Does it make some troubles if also `onPurchaseStoreUpdate` is present in dependency array?

Reply:
no, I only want to execute it when data changes and when data changes the newest version of `onPurchaseStoreUpdate` is used, so no need to add both... this is the reason i don't like that rule within useEffects :D